### DISCUSSION TITLE
Creating a date filter on GridPostArchive

### DISCUSSION THREAD

**Discussion by ben-goodman-uk** [flyntwp/flynt#419] [Q&A] [answered]
I'm trying to create a date dropdown that will allow people to change posts based on date.

The WordPress function wp_get_archives achieves this perfectly by providing a dropdown of dates containing posts.

However I can't see to access or use this function within Flynt or on the archives page.

So far I've tried the following:

- Created a new component to include the dropdown
- Created a shortcode to access the function
- Accessing using `{{ fn('wp_get_archives') }}`
- Accessing using `{{ function('wp_get_archives') }}`
- Recreated archive.php (which worked to override the archives template).
- Recreated a template at `templates/archive.php` (which worked to override the archives template).
- Added `use Timber\Archives;` to `index.php, page.php, single.php, archive.php`
- Added context to all of the above templates `$context['archives'] = new TimberArchives();`
- Tried to implement the Timber\Archives demo as below
- Changed references to items to pull from posts instead

```
<ul>
{% for item in archives.items %}
    <li><a href="{{item.link}}">{{item.name}}</a></li>
    {% for child in item.children %}
        <li class="child"><a href="{{child.link}}">{{child.name}}</a></li>
    {% endfor %}
{% endfor %}
</ul>
```

Every other WordPress function seems to work ok, `{{ fn('site_url') }}`, perfectly.

There seems to be an option within Timber however I can't get this to work either!

https://timber.github.io/docs/reference/timber-archives/

Any help would be greatly appreciated.

**Comment by domtra** [comment, marked as answer]
Hi @ben-goodman-uk ,

I think this should all just work fine. Esp. because all other WordPress functions are working, as well.

Have you tried changing the posts page in wp-admin under Settings -> Reading? Just change the "Your homepage displays" settings save and change it back. This has worked for me before.
